rust/windows-client/src-tauri/src/client.rs

```diff
@@ -139,9 +141,22 @@ pub(crate) fn run() -> Result<()> {
 
 /// `gui::run` but wrapped in `anyhow::Result`
 ///
-/// Error handling can go here soon
+/// Automatically logs or shows error dialogs for important user-actionable errors
 fn run_gui(params: GuiParams) -> Result<()> {
-    Ok(gui::run(params)?)
+    let result = gui::run(params);
+
+    // Make sure errors get logged, at least to stderr
+    if let Err(error) = &result {
+        tracing::error!(?error, "client::run error");
+        native_dialog::MessageDialog::new()
+            .set_title("Firezone Error")
+            .set_text(&format!("{error}"))
```

Comment:
It will probably be a bit more verbose but I think it would be better to not just display print the error. What if we instead match on the concrete error variant and hardcode a message here? Given that it is user-facing, it will be much easier to control the UX that way.